frontend/packages/core/src/Charts/timeseries.tsx

```diff
@@ -0,0 +1,163 @@
+import React from "react";
+import {
+  CartesianGrid,
+  Legend,
+  Line,
+  LineChart,
+  ReferenceLine,
+  ResponsiveContainer,
+  Tooltip,
+  XAxis,
+  YAxis,
+} from "recharts";
+
+import { calculateDomainEdges, calculateTicks, localTimeFormatter } from "./helpers";
+
+export type ReferenceLineAxis = "x" | "y";
+/*
+  For reference lines (dashed lines), you can set the `axis` property to "x" or "y" to denote which axis 
+  the line is on. These can be useful for showing limits or thresholds on graphs.
+  There is already a type called `ReferenceLineProps` in Recharts so we avoid that name
+*/
+export interface TimeseriesReferenceLineProps {
+  axis: ReferenceLineAxis;
+  coordinate: number;
+  color: string;
+  label?: string;
+}
+
+export interface LineProps {
+  dataKey: string;
+  color: string;
+  strokeWidth?: number;
+  animationDuration?: number;
+}
+export interface TimeseriesChartProps {
+  data: any;
+  xAxisDataKey?: string;
+  yAxisDataKey?: string;
+  lines: LineProps[];
+  refLines?: TimeseriesReferenceLineProps[];
+  singleLineMode?: boolean; // if false, the Y Axis will be based off the max and min of all data combined
+  // The assumption is that multiple lines would want to share the same Y Axis.
+  drawDots?: boolean;
+  enableLegend?: boolean;
+  enableGrid?: boolean;
+  tickFormatterFunc?: (timeStamp: number) => string;
+  xDomainSpread?: number | null;
+  yDomainSpread?: number | null;
+  connectNulls?: boolean;
+  regularIntervalTicks?: boolean;
+  // TODO: add ref dots, ref areas, zoom enabled, auto colors,
+  // tooltip options, activeDot options, dark mode / styling,
+}
+
+/*
+  For the lines, use the `dataKey` property to denote which data points belong to that line. Make sure that
+  all the dataKeys match appropriately (the XAxis datakey should correspond to the data that is graphed along
+  the XAxis, and same for Y. Note that we currently set the XAxis to be a time scale, hence the name
+  Timeseries Chart).
+  
+  The data will be internally sorted by the XAxis dataKey when using single line mode. When having multiple lines
+  (singleLineMode is false) then the user is responsible for sorting and grouping the data appropriately. 
+  If they do not sort and group it, there can be discontinuities in the lines. Also, when using multiple lines,
+  the user should pass the yAxisDataKey as the biggest ranged y axis datakey in the data, otherwise data will get chopped off.
+
+  The timestamps are interpreted as unix milliseconds.
+*/
+const TimeseriesChart = ({
+  data,
+  xAxisDataKey = "timestamp",
+  yAxisDataKey = "value",
+  lines,
+  refLines,
+  singleLineMode = true,
+  drawDots = true,
+  enableLegend = true,
+  enableGrid = true,
```

Comment:
nit i would rename these 
```suggestion
  legend = true,
  grid = true,
```